instructor/common.py

```diff
@@ -672,11 +672,34 @@ def western_blot_band_size(request):
         exclude=['primary', 'secondary', 'western_blot']
     )
     if request.method == "POST":
+        weights_by_type = ['wc_weight', 'nuc_weight', 'cyto_weight']
         formset = AntibodiesFormset(request.POST)
         if formset.is_valid():
-            entries = formset.save(commit=False)
-            for form in entries:
-                form.save()
+            antibodies = formset.save(commit=False)
+            strain_treatments = models.StrainTreatment.objects.filter(assignment=assignment)
+            for antibody in antibodies:
+                for index, field in enumerate(weights_by_type):
+                    weight_str = getattr(antibody, field)
+                    weights = [float(s) for s in weight_str.split(',')
```

Comment:
Since this list comprehension is also in `western_blot_band_intensity` can you refactor it into a function shared between both?